block-producer/src/batch_builder/mod.rs

```diff
@@ -94,7 +94,7 @@ where
             locked_ready_batches.drain(..num_batches_in_block).collect()
         };
 
-        match self.block_builder.build_block(batches_in_block.clone()).await {
+        match self.block_builder.build_block(&batches_in_block).await {
             Ok(_) => {
                 // block successfully built, do nothing
             },
```

Comment:
Here, I think it is fine as we don't introduce any extra cloning (block builder does not take ownership of the batches, and just clones relevant pieces from them as needed).